browser-test/src/support/session_timeout.ts

```diff
@@ -0,0 +1,228 @@
+import {Page} from 'playwright'
+import {AdminSettings, loginAsAdmin, logout, validateToastMessage} from '.'
+import {waitForHtmxReady} from './wait'
+import {test, expect} from './civiform_fixtures'
+
+type TimeoutData = {
+  inactivityWarning: number
+  inactivityTimeout: number
+  totalWarning: number
+  totalTimeout: number
+  currentTime: number
+}
+
+type TimeoutType = 'inactivityWarning' | 'totalWarning' | 'totalTimeout'
+
+/**
+ * Gets the adjusted timeout data that accounts for clock skew
+ * This matches what the SessionTimeoutHandler uses internally
+ */
+async function getAdjustedTimeoutData(page: Page): Promise<TimeoutData | null> {
+  return page.evaluate(() => {
+    const cookie = document.cookie
+      .split('; ')
+      .find((row) => row.startsWith('session_timeout_data='))
+    if (!cookie) return null
+
+    try {
+      const value = cookie.split('=')[1]
+      const decoded = atob(decodeURIComponent(value))
+      const data: TimeoutData = JSON.parse(decoded) as TimeoutData
+
+      // Calculate clock skew between client and server (same as in SessionTimeoutHandler)
+      const clientNow = Math.floor(Date.now() / 1000)
+      const clockSkew = clientNow - data.currentTime
+
+      // Return adjusted data
+      return {
+        inactivityWarning: data.inactivityWarning + clockSkew,
+        inactivityTimeout: data.inactivityTimeout + clockSkew,
+        totalWarning: data.totalWarning + clockSkew,
+        totalTimeout: data.totalTimeout + clockSkew,
+        currentTime: data.currentTime,
+      }
+    } catch (e) {
+      console.error('Failed to parse timeout cookie:', e)
+      return null
+    }
+  })
+}
+
+/**
+ * Helper function to mock the Date object and advance time
+ */
+async function mockDateAndAdvanceTime(page: Page, advanceMilliseconds: number) {
```

Comment:
```suggestion

```
(blocking) I would prefer that the [Playwright clock api](https://playwright.dev/docs/clock#introduction) be used instead of a custom solution.